src/uharfbuzz/_harfbuzz.pyx

```diff
@@ -1369,3 +1401,375 @@ def repack_with_tag(tag: str,
+        return self.sets(set_type)
+
+    def __setitem__(self, set_type : SubsetInputSets, iterable) -> Set:
+        s = self.sets(set_type)
+        s.set(iterable)
+
+    @property
+    def flags(self) -> SubsetFlags:
+        cdef unsigned subset_flags = hb_subset_input_get_flags(self._input)
+        return SubsetFlags(subset_flags)
+
+    @flags.setter
+    def flags(self, flags: SubsetFlags) -> None:
+        hb_subset_input_set_flags(self._input, int(flags))
+
+
+cdef class Set:
+    cdef hb_set_t* _hb_set
+
+    INVALID_VALUE = HB_SET_VALUE_INVALID
+
+    def __cinit__(self, init = set()):
+        self._hb_set = hb_set_create()
+        if not hb_set_allocation_successful(self._hb_set):
+            raise MemoryError()
+
+        self.set(init)
+
+    def __dealloc__(self):
+        hb_set_destroy(self._hb_set)
+
+    @staticmethod
+    cdef Set from_ptr(hb_set_t* hb_set):
+        """Create Set from a pointer taking ownership of a it."""
+
+        cdef Set wrapper = Set.__new__(Set)
+        wrapper._hb_set = hb_set
+        return wrapper
+
+    def copy(self) -> Set:
+        c = Set()
+        c._hb_set = hb_set_copy(self._hb_set)
+        return c
+
+    def clear(self):
+        hb_set_clear(self._hb_set)
+
+    def is_empty(self) -> bool:
+        return hb_set_is_empty(self._hb_set)
+
+    def __bool__(self) -> bool:
+        return not self.is_empty()
+
+    def invert(self):
+        hb_set_invert(self._hb_set)
+
+    def has(self, c: int) -> bool:
+        return hb_set_has(self._hb_set, c)
+
+    def __contains__(self, c) -> bool:
+        if type(c) != int: # TODO Small-int?
+            return False
+        return self.has(c)
+
+    def add(self, c: int):
+        hb_set_add(self._hb_set, c)
+        if not hb_set_allocation_successful(self._hb_set):
+            raise MemoryError()
+
+    def add_range(self, first: int, last: int):
+        hb_set_add_range(self._hb_set, first, last)
+        if not hb_set_allocation_successful(self._hb_set):
+            raise MemoryError()
+
+    def remove(self, c: int):
+        if not c in self:
+            raise KeyError, c
+        hb_set_del(self._hb_set, c)
+        if not hb_set_allocation_successful(self._hb_set):
+            raise MemoryError()
+
+    def discard(self, c: int):
+        hb_set_del(self._hb_set, c)
+        if not hb_set_allocation_successful(self._hb_set):
+            raise MemoryError()
+
+    def del_range(self, first: int, last: int):
+        hb_set_del_range(self._hb_set, first, last)
+        if not hb_set_allocation_successful(self._hb_set):
+            raise MemoryError()
+
+    def __delitem__(self, c: int):
+        self.remove(c)
+
+    def is_equal(self, other: Set) -> bool:
+        return hb_set_is_equal(self._hb_set, other._hb_set)
+
+    def __eq__(self, other):
+        if type(other) != Set:
+            return NotImplemented
+        return self.is_equal(other)
+
+    def hash(self) -> int:
```

Comment:
Idiomatically, I think this should be just `__hash__`? `hash` is a builtin.

Reply:
> I'm strongly in favor over only implementing dunder methods for copying, setting, getting, etc., to keep it idiomatic in Python.

Done.

> Also, I'd suggest keeping the `property` getters and setters for SubsetInput, because it allows us to ignore `SubsetInputSets` in the public API :)

I'm not convinced that's a good thing though. But sure, go ahead and add those on top.